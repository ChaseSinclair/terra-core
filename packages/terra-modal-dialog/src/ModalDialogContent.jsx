import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogContent.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogContent = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-content'} {...customProps} >
    {children}
  </div>
  );


ModalDialogContent.propTypes = propTypes;
ModalDialogContent.defaultProps = defaultProps;

export default ModalDialogContent;
