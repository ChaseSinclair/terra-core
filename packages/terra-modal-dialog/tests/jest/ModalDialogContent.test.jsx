import React from 'react';
import ModalDialogContent from '../../src/ModalDialogContent';

const modalDialogContent = <ModalDialogContent>This is content</ModalDialogContent>;

it('should shallow an open modal', () => {
  const modal = shallow(modalDialogContent);
  expect(modal).toMatchSnapshot();
});

it('should mount an open modal', () => {
  const modal = mount(modalDialogContent);
  expect(modal).toMatchSnapshot();
});

it('should render an open modal', () => {
  const modal = render(modalDialogContent);
  expect(modal).toMatchSnapshot();
});

it('should have class terra-ModalDialog-content', () => {
  const modal = shallow(modalDialogContent);
  expect(modal.prop('className')).toEqual('terra-ModalDialog-content');
});
