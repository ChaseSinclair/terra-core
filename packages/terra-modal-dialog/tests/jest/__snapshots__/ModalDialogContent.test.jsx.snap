exports[`test should mount an open modal 1`] = `
<ModalDialogContent>
  <div
    className="terra-ModalDialog-content">
    This is content
  </div>
</ModalDialogContent>
`;

exports[`test should render an open modal 1`] = `
<div
  class="terra-ModalDialog-content">
  This is content
</div>
`;

exports[`test should shallow an open modal 1`] = `
<div
  className="terra-ModalDialog-content">
  This is content
</div>
`;
